ansible_inventory_server/jujurest.py

```diff
@@ -61,14 +61,32 @@
     return status
 
 
+def juju_filter_ip_addresses(data, kwargs):
+    """Filters IP addresses using network interface name and/or subnet"""
+
+    ip_addresses = []
+
+    try:
+        interface_filter = kwargs.get('interface')
+        for interface, interface_data in data['network-interfaces']:
+            if interface_filter and interface != interface_filter:
+                continue
+
+            ip_addresses.extend(interface_data.get('ip-addresses') or [])
+
+    except (KeyError, ValueError):
```

Comment:
Please catch AttributeError as well due to .get(). Please verify that KeyError is needed